src/Context/products-context.js

```diff
@@ -0,0 +1,348 @@
+import { useReducer, useContext, createContext } from "react";
+
+const ProductsContext = createContext(null);
+
+const reducer = (state, action) => {
+  const encodedToken = localStorage.getItem("ENCODED_TOKEN_2");
+
+  switch (action.type) {
+    case "Add to Liked":
+      const likedVideoFound = state.likedArray.find(
+        (video) => video._id === action.payload._id
+      );
+      if (likedVideoFound) {
+        fetch(`/api/user/likes/${action.payload._id}`, {
+          method: "DELETE",
+          headers: {
+            "Content-Type": "application/json",
+            Accept: "application/json",
+            authorization: encodedToken,
+          },
+        })
+          .then((res) => res.json())
+          .then((data) => console.log(data))
+          .catch((err) => console.log(err));
+        return {
+          ...state,
+          likedArray: state.likedArray.filter(
+            (movie) => movie.id !== action.payload.id
+          ),
+        };
+      } else {
+        fetch("/api/user/likes", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Accept: "application/json",
+            authorization: encodedToken,
+          },
+          body: JSON.stringify({ video: action.payload }),
+        })
+          .then((res) => res.json())
+          .then((data) => console.log(data))
+          .catch((err) => console.log(err));
+        return {
+          ...state,
+          likedArray: [...state.likedArray, action.payload],
+          dislikedArray: state.dislikedArray.filter(
+            (movie) => movie.id !== action.payload.id
+          ),
+        };
+      }
```

Comment:
same over here, when you are doing `add to like`, call a function that is doing the network call and dispatching the necessary data from inside it. Directly writing the network call inside the reducer is not a good idea, this might run into some unexpected behavior.